### PR TITLE
obesity epidemic

### PR DESCRIPTION
**Pull request by DylanHsu**
fix smearing for fat jets!

### CODE REVIEW

File: Flat/src/JetsMods.cc [hunk 1]
```diff
@@ -14,9 +14,18 @@ inline float centralOnly(float x, float aeta, float def = -1)
...
 
-JetWrapper BaseJetMod::shiftJet(const Jet& jet, shiftjes shift, bool smear)
+JetWrapper BaseJetMod::shiftJet(const Jet& jet, shiftjes shift, bool smear, bool isFatJet)
 {
-  float pt = smear ? jet.ptSmear : jet.pt();
+  float pt = jet.pt();
+  if (smear) {
+    if (isFatJet) {
+      double smearFac=1, smearFacUp=1, smearFacDown=1;
+      jer.get()->getStochasticSmear(pt,jet.eta(),event.rho,smearFac,smearFacUp,smearFacDown);
```
Comment (sidnarayanan) [inline Flat/src/JetsMods.cc]: You can just do `jer->getStocasticSmear...` here

**Comment by sidnarayanan** [comment]
Good catch - will merge after that one thing with the unique_ptr is changed.

**Comment by DylanHsu** [comment]
done. I have literally never seen a shared pointer before in my life

**Comment by sidnarayanan** [comment]
I renamed `isFatJet` -> `recalcSmear`, since we may want to redo this for AK4 jets too if the smearing gets updated.

**Pull request merged**